### ISSUE TITLE
Xamarin Android Project Using 4.01 Wants the OLD (2 arg) Parameters for LoginAsync?

### ISSUE THREAD

**Issue by dgerding** [Azure/azure-mobile-apps-net-client#372]
I'm getting the error "No overload for method 'LoginAsync' takes 3 arguments" for the code:

```
Wordbahn.FormsApp.App.User = await App.MobileServiceClient.LoginAsync(this,
                        MobileServiceAuthenticationProvider.MicrosoftAccount, FormsApp.Constants.AUTH_GOOGLE_CALLBACK_URL);
```

The problem is, I think, that these _are_ the correct parameters.  

I have gone to the trouble of using the old Android. library, re (https://github.com/Azure/azure-mobile-apps-net-client/issues/361) just in case.  Made no difference.

I have no copies of the older Client library in the solution path or in packages.  I've cleaned all bin/obj folders, restarted etc...

This signature _is_ working for the iOS project.  

Any ideas?

Dave G

**Comment by BioTurboNick** [comment]
This is a duplicate of the issue I reported in #361.

What's happening is that you're installing 4.0.0 into a project with a v23 Android SDK. Because of the dependencies, it references the .Net Standard library instead of the MonoAndroid library. The two methods you are seeing are the cross-platform ones, while the Android-specific ones are only present in the MonoAndroid DLL. If you update to the v25 SDK and support libraries, then reinstall the 4.0.0 Azure.Mobile.Client package, it should give you the right one. You can look at the properties for the reference for the Android project in the Solution Explorer--the path will tell you if it is the netstandard or MonoAndroid one. You can manually remove and add if you need to.

It's not pretty.

**Comment by dgerding** [comment]
@BioTurboNick  Thanks for input.  I had actually misunderstood the thread to suggest that add the 23 version of the custom tabs control was required.  That's the only Xamarin Android nuget targeting 23.  The rest are 25 and the project itself is set to compile against latest (7.1) and target 25.

I'll remove the single 23 component, uninstall 4.01 azure mobile client and then reinstall it.  Please let me know if I'm following your recommendation correctly as I am still stuck. Thanks.

**Comment by BioTurboNick** [comment]
@dgerding - You want 4.0.0, not 4.0.1, but that should work otherwise.

**Comment by dgerding** [comment]
With your help I got it to build.  Seems really "brittle" deploying.  Complains lot about Microsoft.Bcl build being absent even though it isn't absent I had to turn off 

I was under the impression that n.n.x releases, where x changes, are bug fix releases.  Are you recommending against 4.0.1 because you know it won't work or because yours is running on 4.0.0.  I'm 4.0.0 now, too.  Just trying to get my expectations and understanding right.

Thanks again.

**Comment by BioTurboNick** [comment]
Those Bcl.Build warnings are something else I think--I forget how I got rid of them, something to do with different project types referencing each other, but usually spurious?

And yeah, theoretically a point release shouldn't have breaking changes, just fixes. I struggled with this for a good week or so figuring out what was wrong. I know for a fact--and others posting on here confirm--that 4.0.1 is broken.

The record here shows that when v4 was released, someone pointed out that it was a bad experience for it to need v25 support libraries since the default install is v23. So someone downgraded to v23 in this project and submitted a pull request, which was accepted, apparently without proper testing, which was then released as 4.0.1.

As someone else pointed out, theoretically the only reason the downgrade caused a problem was because the internal login code for Android used a function in a way that was cautioned against.
